### PR TITLE
fix(taiko-client): valid status check in `BatchGetBlocksProofStatus`

### LINKED ISSUE #18594: fix(taiko-client): prover will submit unmeaningful contests when aggregation is enabled

### PR DESCRIPTION
**Pull request by YoGhurt111**

### CODE REVIEW

**Review by davidtaikocha — APPROVED** [review]

File: packages/taiko-client/pkg/rpc/utils.go [hunk 1]
```diff
@@ -265,22 +265,19 @@ func BatchGetBlocksProofStatus(
...
-			parent, err := cli.L2.HeaderByNumber(gCtx, new(big.Int).Sub(id, common.Big1))
-			if err != nil {
-				return err
-			}
-			parentHashes[i] = parent.Hash()
-			parents[i] = parent
-			blockIDs[i] = id.Uint64()
+			parentIDs[i] = new(big.Int).Sub(id, common.Big1)
+			blockIDs[i] = id
+			blocks[i] = id.Uint64()
```
Comment (davidtaikocha) [inline packages/taiko-client/pkg/rpc/utils.go:280]: This `blocks` var name is really confusing, let's rename it?

Comment (YoGhurt111) [reply packages/taiko-client/pkg/rpc/utils.go:280]: OK

**Review by davidtaikocha — APPROVED** [review]

File: packages/taiko-client/pkg/rpc/utils.go [hunk 2]
```diff
@@ -265,22 +265,19 @@ func BatchGetBlocksProofStatus(
 	defer cancel()
 	var (
 		parentHashes   = make([][32]byte, len(ids))
-		parents        = make([]*types.Header, len(ids))
-		blockIDs       = make([]uint64, len(ids))
+		parentIDs      = make([]*big.Int, len(ids))
+		blockIDs       = make([]*big.Int, len(ids))
+		uint64BlockIDs = make([]uint64, len(ids))
 		result         = make([]*BlockProofStatus, len(ids))
 		highestBlockID = big.NewInt(0)
```
Comment (mask-pp) [inline packages/taiko-client/pkg/rpc/utils.go:272]: Recommend use `highestBlockID = atomic.Pointer[big.Int]{}`.

File: packages/taiko-client/pkg/rpc/utils.go [hunk 3]
```diff
@@ -290,50 +287,45 @@ func BatchGetBlocksProofStatus(
...
 	if err != nil {
 		return nil, err
 	}
-	highestHeader, err := cli.WaitL2Header(ctxWithTimeout, highestBlockID)
+	_, err = cli.WaitL2Header(ctxWithTimeout, highestBlockID)
 	if err != nil {
 		return nil, err
 	}
-	g, gCtx = errgroup.WithContext(ctxWithTimeout)
+	blockHeaders, err := cli.L2.BatchHeadersByNumbers(ctxWithTimeout, blockIDs)
```
Comment (mask-pp) [inline packages/taiko-client/pkg/rpc/utils.go:310]: check `len(blockHeaders)` must equal to `len(ids)`.

File: packages/taiko-client/pkg/rpc/utils.go (hunk 2 above)
Comment (YoGhurt111) [reply packages/taiko-client/pkg/rpc/utils.go:274]: Done

File: packages/taiko-client/pkg/rpc/utils.go (hunk 3 above)
Comment (YoGhurt111) [reply packages/taiko-client/pkg/rpc/utils.go:308]: Done

**Review by mask-pp — APPROVED** [review]

**Pull request merged**